### PR TITLE
Windows test fixes

### PR DESCRIPTION
**Pull request by gabriel**
- Replacing use of shell scripts with a test go program (which is cross platform)
- Extend some timeouts which can cause flakiness on slow windows machine
- Don't SIGTERM on windows
- Change some EqualError assertions that don't apply on Windows
- --Skip-- Fix symlink tests on Windows
- Skip file permission tests on Windows
- Fix path bugs on Windows

### CODE REVIEW

**Comment by gabriel** [comment]
@keybase/updater-hackers

File: util/file.go [hunk 1]
```diff
@@ -131,6 +134,7 @@ func openTempFile(prefix string, suffix string, mode os.FileMode) (string, *os.F
 
 // FileExists returns whether the given file or directory exists or not
 func FileExists(path string) (bool, error) {
+	fmt.Printf("%s\n", path)
```
Comment (chrisnojima) [inline util/file.go]: 🔥

File: util/file.go [hunk 2]
```diff
@@ -289,3 +293,33 @@ func ReadFile(path string) ([]byte, error) {
...
+
+// URLStringForPath returns an URL as string with file scheme for path.
+// For example,
+//     /usr/local/go/bin => file:///usr/local/go/bin
+//     C:\Go\bin => file:///C:/Go/bin
+func URLStringForPath(path string) string {
+	switch runtime.GOOS {
+	case "windows":
+		// Include leading slash on windows
+		return fmt.Sprintf("%s:///%s", fileScheme, filepath.ToSlash(path))
```
Comment (chrisnojima) [inline util/file.go]: doesn't this need to be escaped? paths with spaces and etc

File: util/file_test.go [hunk 3]
```diff
@@ -25,8 +27,11 @@ func TestNewFile(t *testing.T) {
 
 	fileInfo, err := os.Stat(filename)
 	assert.NoError(t, err)
-	assert.True(t, 0600 == fileInfo.Mode().Perm())
 	assert.False(t, fileInfo.IsDir())
+
+	if runtime.GOOS != "windows" {
+		assert.True(t, 0600 == fileInfo.Mode().Perm())
```
Comment (chrisnojima) [inline util/file_test.go]: does windows have no concept of this or is the value different. if the latter we should check it

File: util/file_test.go [hunk 4]
```diff
@@ -95,14 +106,17 @@ func TestIsDirReal(t *testing.T) {
...
 
-	symLinkPath := TempPath("", "TestIsDirReal")
-	err = os.Symlink(os.TempDir(), symLinkPath)
-	defer RemoveFileAtPath(symLinkPath)
-	assert.NoError(t, err)
-	ok, err = IsDirReal(symLinkPath)
-	assert.Error(t, err)
-	assert.Equal(t, "Path is a symlink", err.Error())
-	assert.False(t, ok)
+	// Windows requires privileges to create symbolic links
```
Comment (chrisnojima) [inline util/file_test.go]: is this true? in our repo we use mklink /j which (i don't think) requires elevated priveledges

File: util/file_test.go [hunk 5]
```diff
@@ -267,5 +281,26 @@ func TestReadFile(t *testing.T) {
 	assert.Equal(t, dataIn, dataOut)
 
 	_, err = ReadFile("/invalid")
-	assert.EqualError(t, err, "open /invalid: no such file or directory")
+	assert.Error(t, err)
+	require.True(t, strings.HasPrefix(err.Error(), "open /invalid: "))
+}
+
+func TestURLStringForPath(t *testing.T) {
+	if runtime.GOOS == "windows" {
```
Comment (chrisnojima) [inline util/file_test.go]: we need to test more complicated paths, like those with weird chars / etc in them

**Comment by chrisnojima** [comment]
@jzila 👀

File: util/file_test.go (hunk 3 above)
Comment (gabriel) [inline util/file_test.go]: yeah windows doesn't have this

File: util/file.go (hunk 2 above)
Comment (gabriel) [inline util/file.go]: Added path escaping and tests

File: util/file_test.go (hunk 4 above)
Comment (gabriel) [inline util/file_test.go]: Fixed the test to do mklink /j on windows

**Comment by gabriel** [comment]
bump

**Pull request merged**